lib/charms/opensearch/v0/helper_charm.py

```diff
@@ -167,16 +167,19 @@ def run_cmd(command: str, args: str = None) -> SimpleNamespace:
         command: can contain arguments
         args: command line arguments
     """
+    command_with_args = command
     if args is not None:
-        command = f"{command} {args}"
+        command_with_args = f"{command} {args}"
 
-    command = " ".join(command.split())
+    command_with_args = " ".join(command_with_args.split())
 
+    # only log the command and no arguments to avoid logging sensitive information
+    command = mask_sensitive_information(command_with_args)
```

Comment:
We def. need unit tests here to show this method works as intended.

Reply:
Yeah I already thought about unit testing this, esp. to make sure nothing is removed from that pattern without a test failing.

Test has been added (incl. special chars), the `opensearch_keystore` lib is covered, I've checked the `run_bin` and `run_cmd` calls there and added the password args to the pattern.